segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/StorageFactoryFactory.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage;
+
+import io.pravega.common.ConfigSetup;
+import java.util.concurrent.ScheduledExecutorService;
+
+public interface StorageFactoryFactory {
```

Comment:
1. This interface has a clunky name.
2. There is no Javadoc explaining what its purpose is.

Reply:
I would really like to see a solution which doesn't require the creation of "Factory Factories". Can we just get away wit passing some lambdas somewhere or configs.

Ideally I wouldn't like to see `ConfigSetup` or implementations of this interface.

Reply:
Renamed `FactoryFactory` to `FactoryCreator`. We have already been using this pattern. This is better defined as an interface as a third party tier2 binding is supposed to implement this. Lambda will be more confusing.